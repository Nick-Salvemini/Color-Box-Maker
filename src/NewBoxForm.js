import React, { useState } from "react";

function NewBoxForm({ addBox }) {
    const INITIAL_STATE = {
        color: '',
        height: '',
        width: ''
    };
    const [formData, setFormData] = useState(INITIAL_STATE);
    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        addBox({ ...formData });
        setFormData(INITIAL_STATE)
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="color">Color</label>
            <input
                id="color"
                type="text"
                name="color"
                value={formData.color}
                placeholder="CSS-friendly Color"
                onChange={handleChange}
            />
            <label htmlFor="height">Height</label>
            <input
                id="height"
                type="text"
                name="height"
                value={formData.height}
                placeholder="Height in Pixels"
                onChange={handleChange}
            />
            <label htmlFor="width">Width</label>
            <input
                id="width"
                type="text"
                name="width"
                value={formData.width}
                placeholder="Width in Pixels"
                onChange={handleChange}
            />
            <button>Add Box</button>
        </form>
    );
}

export default NewBoxForm