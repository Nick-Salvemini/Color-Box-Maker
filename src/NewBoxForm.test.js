import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

describe('NewBoxForm component', () => {
    it('renders without crashing', () => {
        render(<NewBoxForm addBox={() => { }} />);
    });

    it('matches snapshot', () => {
        const { container } = render(<NewBoxForm addBox={() => { }} />);
        expect(container).toMatchSnapshot();
    });

    it('calls addBox function with form data on submission', () => {
        const addBoxMock = jest.fn();
        const { getByLabelText, getByText } = render(<NewBoxForm addBox={addBoxMock} />);

        fireEvent.change(getByLabelText(/color/i), { target: { value: 'blue' } });
        fireEvent.change(getByLabelText(/height/i), { target: { value: '50' } });
        fireEvent.change(getByLabelText(/width/i), { target: { value: '50' } });

        fireEvent.click(getByText(/add box/i));

        expect(addBoxMock).toHaveBeenCalledWith({ color: 'blue', height: '50', width: '50' });
    });
});