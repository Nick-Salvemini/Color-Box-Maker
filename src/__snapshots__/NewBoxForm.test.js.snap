// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBoxForm component matches snapshot 1`] = `
<div>
  <form>
    <label
      for="color"
    >
      Color
    </label>
    <input
      id="color"
      name="color"
      placeholder="CSS-friendly Color"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      name="height"
      placeholder="Height in Pixels"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      name="width"
      placeholder="Width in Pixels"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</div>
`;
